### PR TITLE
[DI][Routing] Fix tracking of globbed resources

### PR DESCRIPTION
**Pull request by nicolas-grekas**
| Q             | A
| ------------- | ---
| Branch?       | 3.4
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | #25904
| License       | MIT
| Doc PR        | -

The current `GlobFileLoader` in `Config` misses resource tracking, so we can't use it and have to use a per-component one instead.

(deps=high failures will be fixed after merging up to master.)

### CODE REVIEW

**Comment by stof** [comment]
New classes should have tests

File: src/Symfony/Component/DependencyInjection/Loader/GlobFileLoader.php [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+class GlobFileLoader extends FileLoader
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function load($resource, $type = null)
+    {
+        foreach ($this->glob($resource, false, $globResource) as $path => $info) {
+            $this->import($path);
+        }
```
Comment (weaverryan) [inline src/Symfony/Component/DependencyInjection/Loader/GlobFileLoader.php]: We don't need any of the parent class's special glob code that's in `import`? I don't fully understand that code - it came from #22985 - but I know you do ;). Just double-checking.

Comment (nicolas-grekas) [reply src/Symfony/Component/DependencyInjection/Loader/GlobFileLoader.php]: The parent extra logic is here to autodetect glob when it's not explicit. Here, it is explicit, so no need.

**Comment by sroze** [comment]
@stof with tests 👍

**Review by nicolas-grekas — COMMENTED** [review]
failure unrelated
thanks @sroze for the tests!

**Comment by nicolas-grekas** [comment]
Votes pending

**Review by weaverryan — APPROVED** [review]

**Comment by nicolas-grekas** [comment]
Thank you @sroze.

**Pull request merged**